File: LLM-Translations/GPT-4/codenet/Python/C/s205112814.c
#include <stdio.h>

int main() {
    int n, m, i;
    scanf("%d %d", &n, &m);

    int gate_cond[m][2], max_under_th = 0, min_upper_th = 100000;
    for (i = 0; i < m; i++) {
        scanf("%d %d", &gate_cond[i][0], &gate_cond[i][1]);
        if (gate_cond[i][0] > max_under_th) {
            max_under_th = gate_cond[i][0];
        }
        if (gate_cond[i][1] < min_upper_th) {
            min_upper_th = gate_cond[i][1];
        }
    }

    if (min_upper_th - max_under_th >= 0) {
        printf("%d\n", min_upper_th - max_under_th + 1);
    } else {
        printf("0\n");
    }

    return 0;
}
// 

File: LLM-Translations/GPT-4/codenet/Python/Go/s205112814.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, m int
	fmt.Scan(&n, &m)

	gateCond := make([][]int, m)
	for i := 0; i < m; i++ {
		var x, y int
		fmt.Scan(&x, &y)
		gateCond[i] = []int{x, y}
	}

	maxUnderTh := math.MinInt32
	minUpperTh := math.MaxInt32
	for _, l := range gateCond {
		if l[0] > maxUnderTh {
			maxUnderTh = l[0]
		}
		if l[1] < minUpperTh {
			minUpperTh = l[1]
		}
	}

	if minUpperTh-maxUnderTh >= 0 {
		fmt.Println(minUpperTh - maxUnderTh + 1)
	} else {
		fmt.Println(0)
	}
}
// 
